pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -702,6 +708,45 @@ func calculateOsqueryPaths(rootDirectory string, registrationId string, runId st
 	return osqueryFilePaths, nil
 }
 
+// cleanUpOldDatabaseLock checks for the presence of a lock file in the given database and removes it
+// if it exists. The lock file can hang around if an osquery process isn't properly terminated, and
+// will prevent us from launching a new osquery process.
+func (i *OsqueryInstance) cleanUpOldDatabaseLock(ctx context.Context, paths *osqueryFilePaths) error {
+	lockFilePath := filepath.Join(paths.databasePath, "LOCK")
+
+	lockFileInfo, err := os.Stat(lockFilePath)
+	if os.IsNotExist(err) {
+		// No lock file, nothing to do here
+		return nil
+	}
+
+	// We do a couple retries here -- Windows complains about removing files in use sometimes.
```

Comment:
My hunch is that this is too aggressive -- if windows is blocking you from removing the file. the osquery probably has it open. I wonder if there's a way we can detect whether osquery is running, or what the state here really is. 

I think forcing the file out, with osquery running will lead to corrupt databases

Reply:
I'd had another idea for handling this --

> The osquery instance and osquery runner both currently have no visibility into the osquery logs -- they're processed by the log adapter, which is entirely separate. However, here and previously (when trying to figure out how to handle launcher falling back to a old version of osquery that is incompatible with the database -- some details https://github.com/kolide/launcher/issues/1937) we've tentatively wanted the osquery instance to be able to respond to particular logs from osquery. Do we want to move log processing into the osquery instance? Or open up some line of communication between the instance and the log adapter?

We could do something like:

- detect via logs that the lock file is in use (like we do when the pidfile is in use)
- gather information about what's using the lock file (like we do when the pidfile is in use)
- as long as it's a dead process, remove the lock file

What do you think?

Reply:
Actually, I could probably update this PR to do the second and third bullet points as-is, without having to mess around w the log adapter 🤔